services/api/service.go

```diff
@@ -2188,34 +2207,26 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		return
 	}
 
-	if contentType == ApplicationOctetStream {
+	if contentType == "application/octet-stream" {
 		log = log.WithField("reqContentType", "ssz")
-		pf.ContentType = "ssz"
-		if err = payload.UnmarshalSSZ(requestPayloadBytes); err != nil {
-			log.WithError(err).Warn("could not decode payload - SSZ")
-
-			// SSZ decoding failed. try JSON as fallback (some builders used octet-stream for json before)
-			if err2 := json.Unmarshal(requestPayloadBytes, payload); err2 != nil {
-				log.WithError(fmt.Errorf("%w / %w", err, err2)).Warn("could not decode payload - SSZ or JSON")
-				api.RespondError(w, http.StatusBadRequest, err.Error())
-				return
-			}
-			log = log.WithField("reqContentType", "json")
-			pf.ContentType = "json"
-		} else {
-			log.Debug("received ssz-encoded payload")
-		}
 	} else {
 		log = log.WithField("reqContentType", "json")
-		pf.ContentType = "json"
-		if err := json.Unmarshal(requestPayloadBytes, payload); err != nil {
-			log.WithError(err).Warn("could not decode payload - JSON")
-			api.RespondError(w, http.StatusBadRequest, err.Error())
-			return
-		}
+	}
+
+	builderEthConsensusVersion := req.Header.Get(HeaderEthConsensusVersion)
+	if builderEthConsensusVersion == "" {
+		// don't reject a builder submission if the Eth-Consensus-Version header is not present
+		builderEthConsensusVersion = api.getForkFromSlot(latestSlot).String()
+		log.Warnf("builder did not send the Eth-Consensus-Version header, using latestSlot %d to infer the fork version: %s", latestSlot, builderEthConsensusVersion)
 	}
```

Comment:
Instead of using the latest slot, I think we should try to use `SubmitBlockRequest.message.slot` (where message is `BidTrace`). This should solve this fork boundary issue; where a builder submits a Fulu block while `latestSlot` is still in Electra. This idea would require partial decoding, should be possible though.

Reply:
<img width="470" height="813" alt="Image" src="https://github.com/user-attachments/assets/47fc2c08-7fa7-4956-a00e-e48fd0eae056" />

Reply:
ya ideally latestSlot would never be in electra, since payload attributes ideally always reflects the latest slot. But partially decoding the payload could be much more accurate.